src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.IDictionary.cs

```diff
@@ -26,9 +25,7 @@ public partial class JsonObject : IDictionary<string, JsonNode?>
         /// </exception>
         public void Add(string propertyName, JsonNode? value)
         {
-            InitializeIfRequired();
-            Debug.Assert(_dictionary != null);
-            _dictionary.Add(propertyName, value);
+            Dictionary.Add(propertyName, value);
```

Comment:
Replaces initialization code with a `Dictionary` property that drives initialization, following the style of the `JsonArray` implementation.